runtime/gc_structs/ClassLocalInterfaceIterator.cpp

```diff
@@ -36,17 +36,16 @@
  * @return the next interface reference slot in the class
  * @return NULL if the class has no more interface references
  */
-J9Class **
+J9Class *
 GC_ClassLocalInterfaceIterator::nextSlot()
 {
-	J9Class **slotPtr;
-
-	if(_iTable == _superclassITable) {
-		return NULL;
+	while (_iTable != _superclassITable) {
+		J9Class *classPtr = _iTable->interfaceClass;
+		_iTable = _iTable->next;
+		if (NULL != classPtr) {
```

Comment:
I'm completely fine with this, but i'll mention that it could be done using 'continue' and only one 'return classPtr;' at the end. Coding standards recommend minimizing mid code returns, but here really it's no better with 'continue'.

Reply:
same for a few other similar loops elsewhere in the code class iterator code

Reply:
I patched it up using a break since there would be one there anyway.